docs: add Decision Record to make event information available (#2084)
eclipse-tractusx/tractusx-edc: docs/development/decision-records/2025-05-22-make-event-info-available/README.md

# Make event information available in basic logging & monitoring stack

## Decision

A small extension (disabled by default) will be added to the TractusX-EDC which subscribes to all internal events and publishes the information directly to an opentelemetry collector instance via http. 

## Rationale

> The Tractus-X EDC has an embedded event mechanism that informs potential listeners about specific incidents, like finalized contract negotiations or ongoing transfers. These events should be routed to a standard logging & monitoring stack to provide the information for further processing, e.g., in Dashboards.

Reviewer: ```suggestion
The Tractus-X connector has an embedded event mechanism that informs potential listeners about specific incidents, like finalized contract negotiations or ongoing transfers. It should be possible to forward these events to logging & monitorings stacks to provide the information for further processing, e.g., for metric calculation or dashboards.
```

> Using an opentelemetry collector instance is also agnostic to the concrete logging & monitoring stack used.

Reviewer: ```suggestion
Using an OpenTelemetry collector instance is also agnostic to the concrete logging & monitoring stack used.
```

## Approach

1. Create a lightweight configurable extension which subscribes to all events.

> 2. By default, the events are published via http directly to the opentelemetry collector, more info see here: https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/README.md

Reviewer: ```suggestion
2. The extension publishes these events via http directly to the OpenTelemetry collector, more info see here: https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/README.md.
```

Reviewer 2: One important note is that _tractusx-edc_ will not, by any means, distribute the mentioned open telemetry collector instance.

> 3. The extension is disabled by default, so that eventual problems with performance of the EDC are not a concern, only if the additional information is really needed

Reviewer 2: @th0masgl It's also important to state in this document that the extension will be shipped with the existing tractusx-edc distributions but will be disabled by default by means of connector configuration - for the reasons you are specifying.

# Make event information available in basic logging & monitoring stack

## Decision

A small extension will be added to the Tractus-X connector which subscribes to all internal events 
and publishes the information directly to an OpenTelemetry collector instance via http.

> The extension can be switched of by configuration. By default it is disabled.

Reviewer 2: ```suggestion
The extension can be switched off/on via configuration. By default it is disabled.
```

## Rationale

The Tractus-X connector has an embedded event mechanism that informs potential listeners about specific incidents, 
like finalized contract negotiations or ongoing transfers. 

> It should be possible to forward these events to logging & monitorings stacks

Reviewer 2: ```suggestion
It should be possible to forward these events a monitoring stacks
```
